domain-management/src/test/java/org/jboss/as/domain/management/security/adduser/AeshJavaConsoleTest.java

```diff
@@ -0,0 +1,134 @@
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this software; if not, write to the Free
+ * Software Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA
+ * 02110-1301 USA, or see the FSF site: http://www.fsf.org.
+ */
+
+package org.jboss.as.domain.management.security.adduser;
+
+import org.aesh.readline.terminal.Key;
+import org.aesh.readline.tty.terminal.TerminalConnection;
+import org.junit.Test;
+
+import java.io.ByteArrayOutputStream;
+import java.io.IOException;
+import java.io.PipedInputStream;
+import java.io.PipedOutputStream;
+import java.nio.charset.Charset;
+
+import static org.junit.Assert.*;
+
+/**
+ * Created by Marek Marusic <mmarusic@redhat.com> on 8/9/17.
+ */
+public class AeshJavaConsoleTest {
+
+    @Test
+    public void testLeftArrow() throws IOException, InterruptedException {
+        AeshJavaConsoleTestResource aeshJavaConsoleTestResource = new AeshJavaConsoleTestResource();
+        aeshJavaConsoleTestResource.start();
+        Thread.sleep(100);
+
+        aeshJavaConsoleTestResource.write("Foo");
+        aeshJavaConsoleTestResource.write(Key.LEFT);
+        aeshJavaConsoleTestResource.write("F");
+        aeshJavaConsoleTestResource.write(Key.ENTER);
+        aeshJavaConsoleTestResource.flushWrites();
+
+        Thread.sleep(100);
+        assertEquals("FoFo", aeshJavaConsoleTestResource.getLine());
+    }
+
+    @Test
+    public void testUpArrow() throws IOException, InterruptedException {
+        AeshJavaConsoleTestResource aeshJavaConsoleTestResource = new AeshJavaConsoleTestResource();
+        aeshJavaConsoleTestResource.start();
+        Thread.sleep(100);
+
+        aeshJavaConsoleTestResource.write("Foo");
+        aeshJavaConsoleTestResource.write(Key.UP);
+        aeshJavaConsoleTestResource.write("F");
+        aeshJavaConsoleTestResource.write(Key.ENTER);
+        aeshJavaConsoleTestResource.flushWrites();
+
+        Thread.sleep(100);
+        assertEquals("FooF", aeshJavaConsoleTestResource.getLine());
+    }
+
+    @Test
+    public void testInterruptionSignal() throws IOException, InterruptedException {
+        AeshJavaConsoleTestResource aeshJavaConsoleTestResource = new AeshJavaConsoleTestResource();
+        aeshJavaConsoleTestResource.start();
+        Thread.sleep(100);
+
+        aeshJavaConsoleTestResource.write("Foo".getBytes());
+        aeshJavaConsoleTestResource.write(Key.CTRL_C.getKeyValuesAsString().getBytes());
+        aeshJavaConsoleTestResource.flushWrites();
+
+        Thread.sleep(100);
+        assertEquals(false, aeshJavaConsoleTestResource.isAlive());
+        assertEquals(Thread.State.TERMINATED, aeshJavaConsoleTestResource.getState());
+    }
+
+    public class AeshJavaConsoleTestResource extends Thread {
```

Comment:
public inner classes are generally considered as not a good practice, is there any reason this class needs to be public? Also please make it static